### PR TITLE
feat(nsis): display "Space required" text for NSIS installer

### PR DESCRIPTION
**Pull request by inickvel**
Implements #7487

Used NSIS `SectionSetSize` function to set correct space.

image

### CODE REVIEW

File: packages/builder-util/src/fs.ts [hunk 1]
```diff
@@ -306,6 +306,25 @@ export function copyDir(src: string, destination: string, options: CopyDirOption
   }).then(() => BluebirdPromise.map(links, it => symlink(it.link, it.file, symlinkType), CONCURRENCY))
 }
 
+export async function dirSize(dirPath: string): Promise<number> {
+  const entries = await readdir(dirPath, { withFileTypes: true })
+
+  const entrySizes = entries.map(async entry => {
+    const entryPath = path.join(dirPath, entry.name)
+
+    if (entry.isDirectory()) return await dirSize(entryPath)
```
Comment (mmaietta) [inline packages/builder-util/src/fs.ts:315]: Please run `pnpm prettier`. That should resolve this. If not, please wrap this in `{ }`
```
if (entry.isDirectory()) {
   return await dirSize(entryPath)
}
```

Please run `pnpm prettier` regardless though :)

File: packages/app-builder-lib/src/targets/nsis/nsisUtil.ts [hunk 2]
```diff
@@ -39,30 +39,44 @@ export const NSIS_PATH = () => {
...
-      infoPromise = this.elevateHelper.copy(appOutDir, target).then(() => target.buildAppPackage(appOutDir, arch))
-      this.archToFileInfo.set(arch, infoPromise)
+      resultPromise = this.elevateHelper
+        .copy(appOutDir, target)
+        .then(() => dirSize(appOutDir))
+        .then(unpackedSize =>
+          target.buildAppPackage(appOutDir, arch).then(fileInfo => ({
+            fileInfo,
+            unpackedSize,
+          }))
```
Comment (mmaietta) [inline packages/app-builder-lib/src/targets/nsis/nsisUtil.ts:67]: `buildAppPackage` seems to modify the `appOutDir` again, after `dirSize` has been called. (Also can't guarantee if that functionality will be changed in the future.)

What are your thoughts on this? It feels more future-proof
```
      resultPromise = this.elevateHelper
        .copy(appOutDir, target)
        .then(() => target.buildAppPackage(appOutDir, arch))
        .then(async fileInfo => ({
            fileInfo,
            unpackedSize: await dirSize(appOutDir),
          }))
```

**Comment by mmaietta** [comment]
Please add a `changeset` via `pnpm generate-changeset` so that this PR gets picked up automatically by the release automation workflow.

**Comment by mmaietta** [comment]
Tests are throwing an error:
```
    Processing script file: "<stdin>" (UTF8)
    warning 6000: unknown variable/constant "{SYSTYPE_PTR}" detected, ignoring (installSection.nsh:31)
```

File: packages/builder-util/src/fs.ts (hunk 1 above)
Comment (inickvel) [reply packages/builder-util/src/fs.ts]: Done

File: packages/app-builder-lib/src/targets/nsis/nsisUtil.ts (hunk 2 above)
Comment (inickvel) [reply packages/app-builder-lib/src/targets/nsis/nsisUtil.ts]: Done, thanks

**Comment by inickvel** [comment]
@mmaietta Fixed all comments. Could you check the PR again? )

**Pull request merged**